src/box/lua/net_box.c

```diff
@@ -57,6 +57,10 @@ netbox_prepare_request(lua_State *L, struct mpstream *stream, uint32_t r_type)
 {
 	struct ibuf *ibuf = (struct ibuf *) lua_topointer(L, 1);
```

Comment:
It seems in the previous commit `iproto_msg` leaks. This simple test makes the connection hang when net_msg_max is reached.
```Lua
netbox = require('net.box')
box.cfg{listen = 3313}
box.schema.user.grant('guest', 'super')

function echo(...)
    return ...
end

c = netbox.connect(box.cfg.listen)
s = c:stream()
for i = 1, 10000 do s:call('echo') end
```

Reply:
Fixed in previous commit (according your recomendation for on_disconnect msg).